### PR TITLE
feat: API payload validation and error handling

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

# TODO 

- [x] Build issue - move `swagger.json` to nocodb package

### CODE REVIEW

File: packages/nocodb/src/lib/meta/api/formViewColumnApis.ts [hunk 1]
```diff
@@ -13,6 +14,7 @@ const router = Router({ mergeParams: true });
 router.patch(
   '/api/v1/db/meta/form-columns/:formViewColumnId',
   metaApiMetrics,
+  // getAjvValidatorMw('swagger.json#/components/schemas/FormViewColumnReq'),
```
Comment (mertmit) [inline packages/nocodb/src/lib/meta/api/formViewColumnApis.ts:17]: We should comment out after adding FormViewColumnReq schema

File: packages/nocodb/src/lib/meta/api/gridViewColumnApis.ts [hunk 2]
```diff
@@ -22,6 +23,7 @@ router.get(
 router.patch(
   '/api/v1/db/meta/grid-columns/:gridViewColumnId',
   metaApiMetrics,
+  getAjvValidatorMw('swagger.json#/components/schemas/GridViewColumnReq'),
```
Comment (mertmit) [inline packages/nocodb/src/lib/meta/api/gridViewColumnApis.ts:26]: GridViewColumnReq schema is missing.

File: packages/nocodb-sdk/src/lib/Api.ts [hunk 3]
```diff
@@ -999,12 +1136,7 @@ export class Api<
      * @response `200` `void` OK
      * @response `401` `void` Unauthorized
      */
-    passwordForgot: (
-      data: {
-        email?: string;
-      },
-      params: RequestParams = {}
-    ) =>
+    passwordForgot: (data: ForgotPasswordReqType, params: RequestParams = {}) =>
```
Comment (wingkwong) [inline packages/nocodb-sdk/src/lib/Api.ts:1139]: prefer to call `PasswordForgotRedType` following other patterns.

**Review by wingkwong — COMMENTED** [review]
- run `npm run fix` on backend
- there are some APIs which are not covered in this PR. If they are meant to be excluded in this PR, it'd better to add a TODO list.

**Comment by pranavxc** [comment]
> * run `npm run fix` on backend
> * there are some APIs which are not covered in this PR. If they are meant to be excluded in this PR, it'd better to add a TODO list.

Done

**Review by wingkwong — COMMENTED** [review]
I think we should unify `meta`. I see some are using `any`, some are using `object | null | string`, some are using `string | object` ...

**Comment by pranavxc** [comment]
> I think we should unify `meta`. I see some are using `any`, some are using `object | null | string`, some are using `string | object` ...

Agree I will use a common type for all the meta :+1:

File: packages/nocodb-sdk/package.json [hunk 4]
```diff
@@ -31,7 +31,7 @@
     "test:prettier": "prettier \"src/**/*.ts\" --list-different",
     "test:spelling": "cspell \"{README.md,.github/*.md,src/**/*.ts}\"",
     "watch:build": "tsc -p tsconfig.json -w",
-    "generate:sdk": "npx --yes swagger-typescript-api@10.0.3 -r -p ../../scripts/sdk/swagger.json -o ./src/lib/  --axios --unwrap-response-data  --module-name-first-tag --type-suffix=Type --templates ../../scripts/sdk/templates"
+    "generate:sdk": "npx --yes swagger-typescript-api@10.0.3 -r -p ../../packages/nocodb/src/schema/swagger.json -o ./src/lib/  --axios --unwrap-response-data  --module-name-first-tag --type-suffix=Type --templates ../../scripts/sdk/templates"
```
Comment (wingkwong) [inline packages/nocodb-sdk/package.json:34]: `../nocodb/src/schema/swagger.json`

File: packages/nocodb/src/lib/models/Filter.ts [hunk 5]
```diff
@@ -55,7 +55,7 @@ export default class Filter {
   value?: string;
 
   logical_op?: string;
-  is_group?: boolean;
+  is_group?: boolean | number;
```
Comment (wingkwong) [inline packages/nocodb/src/lib/models/Filter.ts:58]: `BoolType`?

**Review by wingkwong — COMMENTED** [review]
may take care of the type of `created_at`, `updated_at` as well. currently they are not aligned across files.

**Comment by pranavxc** [comment]
> may take care of the type of created_at, updated_at as well. currently they are not aligned across files.

Will do it an followup PR.

**Review by wingkwong — APPROVED** [review]

**Pull request merged**